### PR TITLE
SoundWire: add Intel NUC M15 support

### PR DESCRIPTION
**Pull request by plbossart**
This should fix the issue reported by @gigago at https://github.com/thesofproject/linux/issues/3049

### CODE REVIEW

**Comment by bardliao** [comment]
LGTM. let's wait for @gigago to check the JD source.

**Review by RanderWang — APPROVED** [review]
LGTM.  @libinyang  and @kv2019i  can you help to make sure it will not happen on ADL ? As I know ADL still uses RT700 by default.

**Comment by plbossart** [comment]
updated commit messages, comments and DMI string to reflect that LAPBC710 and LAPBC510 are the same audio design.

**Review by bardliao — APPROVED** [review]

File: sound/soc/intel/boards/sof_sdw.c [hunk 1]
```diff
@@ -178,6 +178,17 @@ static const struct dmi_system_id sof_sdw_quirk_table[] = {
 					SOF_SDW_PCH_DMIC |
 					RT711_JD2),
 	},
+	{
+		/* NUC15 'Bishop County' LAPBC510 and LAPBC710 skews */
+		.callback = sof_sdw_quirk_cb,
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Intel(R) Client Systems"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "LAPBC"),
```
Comment (kv2019i) [inline sound/soc/intel/boards/sof_sdw.c:186]: Hmm. First I thought this is a bit broad pattern, but it seems all the LAPBC skews share the design, so seems safe.

**Review by kv2019i — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

File: sound/soc/intel/boards/sof_sdw.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c:186]: yes, I checked the pattern and the 'LAP' identifies a product type (LAPtop) and 'BC' is the product name, so no need to check the rest.

**Pull request merged**